Makefile

```diff
@@ -50,11 +50,9 @@ lint: ## Run linter over the codebase
 	"$(GOBIN)/gometalinter" ./pkg/... ./cmd/... ./integration/...
 
 .PHONY: test
-test: generate ## Run unit test (and re-generate code under test)
+test: ## Run unit test (and re-generate code under test)
 	$(MAKE) lint
-	git diff --exit-code pkg/nodebootstrap/assets.go > /dev/null || (git --no-pager diff pkg/nodebootstrap/assets.go; exit 1)
-	git diff --exit-code ./pkg/eks/mocks > /dev/null || (git --no-pager diff ./pkg/eks/mocks; exit 1)
-	git diff --exit-code ./pkg/addons/default > /dev/null || (git --no-pager diff ./pkg/addons/default; exit 1)
+	$(MAKE) generate-aws-mocks-test generate-bindata-assets-test generate-kubernetes-types-test
```

Comment:
@martina-if  @errordeveloper Making `generate-kubernetes-types-test` here causes the code generation (`generate-kubernetes-types`) to happens twice (both here and in `build`).

I think it can be done so that it's only invoked only once, either by removing the duplicated dependency or by depending on the generated type,

Also I think it's bad practice to invoke make from make, that's what dependencies are for.

Reply:
Ok, let's fix it. The thing is that dependencies get invoked in parallel if all are phony, so this is is just a lazy option... Maybe we can do something without declaring all the files as dependencies?

Reply:
We don't need to declare all the files, just the `zz_deepcopy.go` one I think.

Can we also use explicit dependencies instead of invoking make. Pretty pretty please? :)

Reply:
Sure! But what I was trying to say is that when you have phony targets listed as dependencies make in parallel mode will call them all at once and may not get it right, so that was reason I called make directly... It is not a big deal, there is little use of parallel mode for us right now, go build does most of the work.

Reply:
@errordeveloper where did you resolve this?

Reply:
It wasn't fix, I just clicked 'resolve', as in - let's go ahead and fix it per proposal.